Approximate_convex_decomposition/examples/Approximate_convex_decomposition/approx_convex_decomposition_SM_example.cpp

```diff
@@ -0,0 +1,69 @@
+#include <CGAL/approx_decomposition.h>
+#include <CGAL/Surface_mesh.h>
+
+#include <iostream>
+#include <fstream>
+#include <chrono>
+
+typedef CGAL::Exact_predicates_inexact_constructions_kernel Kernel;
+
+typedef CGAL::Surface_mesh<Kernel::Point_3> Mesh;
+
+typedef boost::graph_traits<Mesh>::face_descriptor face_descriptor;
+typedef boost::graph_traits<Mesh>::face_iterator   face_iterator;
+
+int main()
+{
+    // read mesh
+    Mesh mesh;
+    
+    std::ifstream input("data/cube.off");
+//    std::ifstream input("data/teapot.off");
+//    std::ifstream input("data/sword.off");
+//    std::ifstream input("data/cactus.off");
+//    std::ifstream input("data/cheese.off");
+//    std::ifstream input("data/lion-head.off");
+//    std::ifstream input("data/elephant.off");
+    
+    if (!input || !(input >> mesh))
+    {
+        std::cout << "Failed to read mesh" << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    if (CGAL::is_empty(mesh) || !CGAL::is_triangle_mesh(mesh))
+    {
+        std::cout << "Input mesh is invalid" << std::endl;
+        return EXIT_FAILURE;
+    }
+
+    // create property map for cluster-ids
+    typedef std::map<face_descriptor, int> Facet_int_map;
+    Facet_int_map facet_map;
+    boost::associative_property_map<Facet_int_map> facet_property_map(facet_map);
```

Comment:
the point of using Surface_mesh is that you can have some internal properties. See `add_property_map` in the doc.